micrometer/src/test/java/org/jboss/eap/qe/micrometer/prometheus/MicrometerPrometheusTestCase.java

```diff
@@ -0,0 +1,272 @@
+import org.jboss.eap.qe.microprofile.common.setuptasks.MicrometerServerConfiguration;
+import org.jboss.eap.qe.microprofile.common.setuptasks.OpenTelemetryServerConfiguration;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianContainerProperties;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.arquillian.ArquillianDescriptorWrapper;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.client.jaxrs.engines.ApacheHttpClientEngine;
+import org.junit.AfterClass;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Test for prometheus micrometer registry.
+ */
+@RunWith(Arquillian.class)
+public class MicrometerPrometheusTestCase {
+    private static OnlineManagementClient client = null;
+    ArquillianContainerProperties server = new ArquillianContainerProperties(
+            ArquillianDescriptorWrapper.getArquillianDescriptor());
+    private static final String GET_LAST_LOGS_CLI_COMMAND = "/subsystem=logging/log-file=server.log:read-log-file(lines=40)";
+
+    private static boolean serverTypeCheck() {
+        return System.getProperty("jboss.home").toLowerCase().contains("eap");
+    }
+
+    @Before
+    public void testServerTypeCheck() {
+        Assume.assumeTrue(serverTypeCheck());
+    }
+
+    /**
+     * Enable micrometer and prometheus registry before first test execution
+     */
+    @BeforeClass
+    public static void enableMicrometer() throws Exception {
+        if (!serverTypeCheck()) {
+            return;
+        }
+        client = ManagementClientProvider.onlineStandalone();
+        MicrometerServerConfiguration.enableMicrometer(client, null, true);
+        MicrometerPrometheusSetup.enable(client);
+    }
+
+    /**
+     * Disable micrometer and prometheus registry after last test execution
+     */
+    @AfterClass
+    public static void disableMicrometer() throws Exception {
+        if (!serverTypeCheck()) {
+            return;
+        }
+        try {
+            MicrometerServerConfiguration.disableMicrometer(client, true);
+            MicrometerPrometheusSetup.disable(client);
+        } finally {
+            client.close();
+        }
+    }
+
+    /**
+     * Get prometheus metrics from unsecured end-point, check that JVM/System metrics are in the list
+     */
+    @Test
+    public void basicPrometheusTest() throws Exception {
+        MatcherAssert.assertThat(
+                "There seems to be conflict between micrometer prometheus end-point and some another end-point.",
+                client.execute(GET_LAST_LOGS_CLI_COMMAND).stringListValue(),
```

Comment:
I am not against this way of inspecting the logs, but this test suite has some internal APIs to do that, see https://github.com/jboss-eap-qe/eap-microprofile-test-suite/blob/master/tooling-server-configuration/src/main/java/org/jboss/eap/qe/microprofile/tooling/server/log/ModelNodeLogChecker.java#L19

Here's a usage example: https://github.com/jboss-eap-qe/eap-microprofile-test-suite/blob/master/microprofile-open-api/src/test/java/org/jboss/eap/qe/microprofile/openapi/advanced/misc/DisablePropertiesTest.java#L163-L166

WDYT?

Reply:
+1, updated, thank you!